### PR TITLE
Arc updates

### PR DESCRIPTION
**Pull request by jamesstout**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Fixed numerous compiler and analyzer warnings
Fixed crash on Export screen

Where has this been tested?
---------------------------
**Devices/Simulators:** iMac19,1

**macOS Version:**: 10.15.7 (19H2)

**Sequel-Ace Version:** arc_updates latest

### CODE REVIEW

File: Source/Other/CategoryAdditions/SPMutableArrayAdditions.h [hunk 1]
```diff
@@ -38,5 +38,7 @@
 @interface NSMutableArray (SPMutableArrayAdditions)
 
 - (void)reverse;
+- (nullable id)SPsafeObjectAtIndex:(NSUInteger)idx;
```
Comment (Kaspik) [inline Source/Other/CategoryAdditions/SPMutableArrayAdditions.h:41]: let's remove the `SP` prefix for both and use just `safeObjectAtIndex` and `safeAddObject` :)

File: Source/Other/CategoryAdditions/SPNSMutableDictionaryAdditions.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface NSMutableDictionary (SPNSMutableDictionaryAdditions)
+
+
+- (id)SPsafeObjectForKey:(id)key;
```
Comment (Kaspik) [inline Source/Other/CategoryAdditions/SPNSMutableDictionaryAdditions.h:16]: ^^

File: Source/Other/DebuggingSupport/SPLogger.m [hunk 3]
```diff
@@ -160,7 +160,7 @@ - (void)dumpLeaks
 				}
 			}
 			
-			free(&files);
+//			free(&files);
```
Comment (Kaspik) [inline Source/Other/DebuggingSupport/SPLogger.m:163]: remove?

**Review by Kaspik — APPROVED** [review]
Nice! the TODOs and Leaks can be fixed from my branch after merging this, but the renaming would be nice :)

File: Source/Other/DebuggingSupport/SPLogger.m (hunk 3 above)
Comment (jamesstout) [reply Source/Other/DebuggingSupport/SPLogger.m:163]: yes

File: Source/Other/CategoryAdditions/SPMutableArrayAdditions.h (hunk 1 above)
Comment (jamesstout) [reply Source/Other/CategoryAdditions/SPMutableArrayAdditions.h]: done

File: Source/Other/CategoryAdditions/SPMutableArrayAdditions.h [hunk 4]
```diff
@@ -38,7 +38,7 @@
 @interface NSMutableArray (SPMutableArrayAdditions)
 
 - (void)reverse;
-- (nullable id)SPsafeObjectAtIndex:(NSUInteger)idx;
+- (nullable id)safeObjectAtIndex:(NSUInteger)idx;
 - (void)SPsafeAddObject:(nullable id)obj;
```
Comment (Kaspik) [inline Source/Other/CategoryAdditions/SPMutableArrayAdditions.h:42]: remove the prefix too please :)

File: Source/Other/CategoryAdditions/SPNSMutableDictionaryAdditions.h [hunk 5]
```diff
@@ -13,9 +13,9 @@ NS_ASSUME_NONNULL_BEGIN
 @interface NSMutableDictionary (SPNSMutableDictionaryAdditions)
 
 
-- (id)SPsafeObjectForKey:(id)key;
-- (void)SPsafeSetObject:(id)obj forKey:(id)key;
-- (void)SPsafeRremoveObjectForKey:(nullable id)key;
+- (id)safeObjectForKey:(id)key;
```
Comment (Kaspik) [inline Source/Other/CategoryAdditions/SPNSMutableDictionaryAdditions.h:16]: Just a check (not sure) - isn't `key` in Foundation method String, not id?

**Review by Kaspik — COMMENTED** [review]
Couple of notes :)

File: Source/Controllers/DataControllers/SPDatabaseStructure.m [hunk 6]
```diff
@@ -286,12 +286,13 @@ - (void)queryDbStructureWithUserInfo:(NSDictionary *)userInfo
 #endif
 
 		// Delete all stored data for the database to be updated, leaving the structure key
-		[queriedStructure removeObjectForKey:db_id];
+		[queriedStructure safeRremoveObjectForKey:db_id];
```
Comment (Jason-Morcos) [inline Source/Controllers/DataControllers/SPDatabaseStructure.m:289]: I assume this should be remove and nor rremove?

File: Source/Controllers/DataControllers/SPDatabaseStructure.m [hunk 7]
```diff
@@ -286,12 +286,13 @@ - (void)queryDbStructureWithUserInfo:(NSDictionary *)userInfo
...
 		NSPredicate *predicate = [NSPredicate predicateWithFormat:@"NOT SELF BEGINSWITH %@", [NSString stringWithFormat:@"%@%@", db_id, SPUniqueSchemaDelimiter]];
 		[queriedStructureKeys filterUsingPredicate:predicate];
 
 		// Set up the database as an empty mutable dictionary ready for tables, and store a reference
 		[queriedStructure setObject:[NSMutableDictionary dictionary] forKey:db_id];
+		[queriedStructure safeSetObject:[NSMutableDictionary dictionary] forKey:db_id];
```
Comment (Jason-Morcos) [inline Source/Controllers/DataControllers/SPDatabaseStructure.m:295]: Is it supposed to be doing both set and safeSet or should the non safe one be removed?

File: Source/Other/CategoryAdditions/SPNSMutableDictionaryAdditions.h (hunk 5 above)
Comment (jamesstout) [reply Source/Other/CategoryAdditions/SPNSMutableDictionaryAdditions.h:16]: it's `(KeyType <NSCopying>)` ... I couldn't figure out how to use it...

File: Source/Other/CategoryAdditions/SPMutableArrayAdditions.h (hunk 4 above)
Comment (jamesstout) [reply Source/Other/CategoryAdditions/SPMutableArrayAdditions.h]: oops .. sorry missed these. Done.

File: Source/Controllers/DataControllers/SPDatabaseStructure.m (hunk 6 above)
Comment (jamesstout) [reply Source/Controllers/DataControllers/SPDatabaseStructure.m:289]: ahhhh

File: Source/Controllers/DataControllers/SPDatabaseStructure.m (hunk 7 above)
Comment (jamesstout) [reply Source/Controllers/DataControllers/SPDatabaseStructure.m:295]: um yes

**Review by Kaspik — APPROVED** [review]

**Pull request merged**